handlers/userHandler.js

```diff
@@ -0,0 +1,64 @@
+const User = require("../model/userModel");
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const Logging = require('../library/Logging');
+
+const  registerUser = async (req,res) => {
+	const {username, name, email, password} = req.body;
+
+	try {
+		// check if the username already exists
+		const existingUser = await User.findOne({username});
+
+		if(existingUser){
+			return res.status(409).json({error:'username alreasy exists'})
+		}
+
+		// creating a new user
+		const newUser = new User({username, name, email, password});
+
+		// saving newUser to DB
+		await newUser.save();
+
+		// accessing generated userId
+		const userId = newUser._id;
+		Logging.info('userID:', userId);
+
+		res.status(201).json({message:'user successfully registered'});
+	}	catch(error) {
+		res.status(500).jsong({error:'an error occured while registering the user'})
```

Comment:
```suggestion
		res.status(500).json({error:'an error occured while registering the user'})
```
Do we have anything like `jsong` please let me know.